ewms_pilot/utils.py

```diff
@@ -0,0 +1,75 @@
+"""Common utilities."""
+
+
+from typing import Any, Callable, Coroutine, TypeVar
+
+import htchirp  # type: ignore[import]
+from typing_extensions import ParamSpec
+
+from .config import LOGGER
+
+T = TypeVar("T")
+P = ParamSpec("P")
+
+
+def chirp(message: str) -> None:
+    """Invoke HTChirp, AKA send a status message to Condor."""
+    try:
+        chirper = htchirp.HTChirp()
+    except ValueError:  # ".chirp.config must be present or you must provide a host and port
+        return
+
+    with chirper as c:
+        LOGGER.info(f"chirping as '{c.whoami()}'")
+        c.set_job_attr("EWMSPilotProcessing", "True")
+        if message:
+            c.ulog(message)
```

Comment:
I'll note that the `ulog` adds the message to the condor event log, which we currently don't read or save.  It's probably better to use `set_job_attr` for anything you care about, since then it gets saved to the job classads and will live in condor history.